provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/location/LocationController.java

```diff
@@ -1,3 +1,445 @@
+            .providerResponse(ProviderResponse.builder().build())
+            .providerServicesResponse(ProviderServicesResponse.builder().build());
+    String identifier = parameters.getFirst("identifier");
+    ProviderServicesResponse providerServicesResponse = ppmsClient.providerServicesById(identifier);
+    locationWrapper.providerResponse(
+        ProviderResponse.builder()
+            .value(
+                singletonList(
+                    ProviderResponse.Value.builder()
+                        .providerIdentifier(
+                            Integer.parseInt(parameters.get("identifier").toArray()[0].toString()))
+                        .build()))
+            .build());
+    if (providerServicesResponse.value().isEmpty()) {
+      locationWrapper.providerResponse(ppmsClient.providersForId(identifier));
+      locationWrapper.careSitesResponse(ppmsClient.careSitesById(identifier));
+    }
+    return singletonList(
+        locationWrapper.providerServicesResponse(providerServicesResponse).build());
+  }
+
+  private List<LocationWrapper> searchName(MultiValueMap<String, String> parameters) {
+    LocationWrapper.LocationWrapperBuilder locationWrapper =
+        LocationWrapper.builder()
+            .careSitesResponse(CareSitesResponse.builder().build())
+            .providerResponse(ProviderResponse.builder().build())
+            .providerServicesResponse(ProviderServicesResponse.builder().build());
+    List<LocationWrapper> filteredResults = new ArrayList<>();
+    totalRecords = 1;
+    String name = parameters.get("name").toArray()[0].toString();
+    locationWrapper.providerResponse(ppmsClient.providersForName(trimIllegalCharacters(name)));
+    totalRecords = locationWrapper.build().providerResponse().value().size();
+    int page = Integer.parseInt(parameters.getOrDefault("page", singletonList("1")).get(0));
+    int count = Integer.parseInt(parameters.getOrDefault("_count", singletonList("15")).get(0));
+    int fromIndex =
+        Math.min((page - 1) * count, locationWrapper.build().providerResponse().value().size());
+    int toIndex =
+        Math.min((fromIndex + count), locationWrapper.build().providerResponse().value().size());
+    List<ProviderServicesResponse> providerServicesResponsePages =
+        IntStream.range(fromIndex, toIndex)
+            .parallel()
+            .mapToObj(
+                i -> {
+                  try {
+                    return ppmsClient.providerServicesByName(
+                        trimIllegalCharacters(
+                            locationWrapper.build().providerResponse().value().get(i).name()));
+                  } catch (Exception e) {
+                    return ProviderServicesResponse.builder().build();
+                  }
+                })
+            .collect(Collectors.toList());
+    List<ProviderResponse.Value> providerResponsePages =
+        locationWrapper.build().providerResponse().value().subList(fromIndex, toIndex);
```

Comment:
This subset of the providers ties into the retrieval of the provider services above. So this should go first, and then the above parallel stream can be simplified a little:
```
        providerResponsesPaged
            .parallelStream()
            .map(
                prv -> {
                  try {
                    return ppmsClient.providerServicesByName(trimIllegalCharacters(prv.name()));
                  } catch (Exception e) {
                      // ...
                  }
                })
            .collect(Collectors.toList());
```